services/autoscaling/src/simcore_service_autoscaling/utils/utils_docker.py

```diff
@@ -540,9 +541,63 @@ def get__new_node_docker_tags(
     )
 
 
-def is_node_ready_and_available(node: Node, availability: Availability) -> bool:
+def is_node_ready_and_available(node: Node, *, availability: Availability) -> bool:
     assert node.Status  # nosec
     assert node.Spec  # nosec
     return bool(
         node.Status.State == NodeState.ready and node.Spec.Availability == availability
     )
+
+
+_OSPARC_SERVICE_READY_LABEL_KEY: Final[DockerLabelKey] = parse_obj_as(
+    DockerLabelKey, "osparc-services-ready"
+)
+
+
+def is_node_osparc_ready(node: Node) -> bool:
+    if not is_node_ready_and_available(node, availability=Availability.active):
+        return False
+    assert node.Spec  # nosec
+    return bool(
+        node.Spec.Labels
+        and _OSPARC_SERVICE_READY_LABEL_KEY in node.Spec.Labels
+        and node.Spec.Labels[_OSPARC_SERVICE_READY_LABEL_KEY] == "true"
+    )
+
+
+async def set_node_osparc_ready(
+    app_settings: ApplicationSettings,
+    docker_client: AutoscalingDocker,
+    node: Node,
+    *,
+    ready: bool,
+) -> Node:
+    assert node.Spec  # nosec
+    new_tags = deepcopy(cast(dict[DockerLabelKey, str], node.Spec.Labels))
+    new_tags[_OSPARC_SERVICE_READY_LABEL_KEY] = "true" if ready else "false"
+    # NOTE: docker drain sometimes impeed on performance when undraining see https://github.com/ITISFoundation/osparc-simcore/issues/5339
+    available = app_settings.AUTOSCALING_LABELIZE_DRAINED_NODES or ready
+    return await tag_node(
+        docker_client,
+        node,
+        tags=new_tags,
+        available=available,
+    )
+
+
+async def attach_node(
+    app_settings: ApplicationSettings,
+    docker_client: AutoscalingDocker,
+    node: Node,
+    *,
+    tags: dict[DockerLabelKey, str],
+) -> Node:
+    assert node.Spec  # nosec
+    current_tags = cast(dict[DockerLabelKey, str], node.Spec.Labels or {})
+    new_tags = current_tags | tags | {_OSPARC_SERVICE_READY_LABEL_KEY: "false"}
+    return await tag_node(
+        docker_client,
+        node,
+        tags=new_tags,
+        available=app_settings.AUTOSCALING_LABELIZE_DRAINED_NODES,  # NOTE: full drain sometimes impeed on performance
```

Comment:
Chatty said: The spelling "impeed" is incorrect; the correct spelling is "impede." To "impede" means to hinder, obstruct, or slow the progress of something. It conveys a similar sense to "empêcher" in French, indicating the act of making it difficult for something to advance or proceed.

:-)

Reply:
thanks chatty... will fix it